tests/ceph_volume/test_ceph_volume_prepare_lvm.py

```diff
@@ -0,0 +1,74 @@
+import json
+import time
+
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.operations import wait_for_osd_daemon_state
+from cli.utilities.utils import create_yaml_config
+from utility.lvm_utils import lvm_create, pvcreate, vgcreate
+
+
+def _identify_disks(node):
+    """
+    Identifies the disks associated with the node
+    Args:
+        node (ceph): Node in which the cmd is executed
+
+    Returns (list): List of available disks
+    """
+
+    disks = []
+    conf = {"format": "json"}
+    available_disks = json.loads(CephAdm(node).ceph.orch.device.ls(**conf))
+    for key in available_disks:
+        if key.get("devices"):
+            for dev in key.get("devices"):
+                disks.append(dev.get("path"))
+    return list(set(disks))
+
+
+def run(ceph_cluster, **kw):
+    """Create OSDs with pre-created LVMs"""
+
+    # Get configs
+    config = kw.get("config")
+
+    # Get the installer and OSD nodes
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Identify available devices on node
+    disks = _identify_disks(installer)
+
+    # Get required host from config specs
+    nodes = config.get("specs", {}).get("placement", {}).get("hosts")
+    host = [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
+
+    # Create spec file specific to hostname, and devices
+    specs = config.get("specs", {})
+    specs["placement"]["hosts"] = host
+
+    # Create lvm manually
+    lv_sizes = {"data_devices": "4G", "db_devices": "2G", "wal_devices": "2G"}
+    pvcreate(installer, disks[0])
+    vgcreate(installer, "vg_disk_pool", disks[0])
+
+    # Create and update LVM in spec
+    for name, size in lv_sizes.items():
+        lvm_name = lvm_create(installer, name, "vg_disk_pool", size)
+        specs[name]["paths"] = lvm_name
+
+    # Create a spec file after updating devices
+    file = create_yaml_config(installer, specs)
+
+    # Create OSDs with spec file
+    c = {"pos_args": [], "input": file}
+    CephAdm(nodes=installer, mount=file).ceph.orch.apply(**c)
+
+    # Sleep for 60s for OSDs to be generate
+    time.sleep(60)
```

Comment:
instead of contant sleep, check for OSD status.

Reply:
Yes,I have used it just bellow that wait_for_osd_daemon_state method. 

Just that when creating OSDs for the very first time; ceph osd ls won't list anything until OSD creation takes place.